app/src/main/java/org/mozilla/focus/home/NavigationOverlayFragment.kt

```diff
@@ -54,6 +54,8 @@ class NavigationOverlayFragment : Fragment() {
             }
         }
 
+        overlay.initialHomescreenBackground.visibility = if (isInitialHomescreen) View.VISIBLE else View.GONE
```

Comment:
It's possible that I'm misunderstanding, but if we are controlling the overlay show/hide via visibility, why did we need to break it out into a fragment?

Reply:
We're dynamically changing the two layouts to support two different modes the fragment can be added in:
- Initial homescreen (where there is no semi opaque background and the url is usable)
- "Dialog" (where there is a semi opaque background and the url bar is obscured)

For the initial homescreen mode, we have, layered bottom to top:
- Grey background (the line being changed) that fills in the space where the WebView is
- The slide up view shadow
- The slide up view background (with rounded corners). This is the same color as the grey background. Without the grey background, because we cut out the corners, the user would see the white WebView or the garbage graphics buffer, if we don't have a view there (I can't remember which)
- The slide up view home tile content

I see that the comments in this file are inadequate:
- They don't explain what the views do (especially this one)
- They don't explain which views are used when in which modes

I wrote them like this to avoid redundancy and make the code the single source of truth but I can see that that makes this layout file really confusing (here's to defining layouts in one code file with all dynamic changes!). **I'll update the comments.**

Reply:
For background context, I put the overlay into a Fragment to:
- Better encapsulate the overlay functionality
- Make it easier change the UX to show/hide the fragment at different times (fragments, with their encapsulated logic, are much easier to remove/add than regular views).